Automattic/wp-calypso
# [WordPress → Jetpack app branding] Update app promo in Reader

_Context:_ Following the plans to [refocus the WordPress apps on core features](https://make.wordpress.org/mobile/2022/07/27/refocusing-the-wordpress-app-on-core-features/), we plan to move features that are specific to WordPress.com to the Jetpack app. As such, we will eventually replace all mentions of the WordPress app in Calypso with Jetpack.

#### Proposed Changes

The main focus of this PR involves updating the mobile app promo in the Reader's sidebar to mention the Jetpack, rather than the WordPress, mobile app. This also has the side effect of slightly changing the desktop app promo's appearance, as a shared component is used to promote both the desktop and mobile.   

<details>
 <summary>Code notes ⤵️</summary>

* The `jetpack/app-branding` feature flag (added in https://github.com/Automattic/wp-calypso/pull/66814) is used to only display the new Jetpack app mentions when the flag is set to true (currently only for wpcalypso and development builds).
* Note, the feature flag, along with any code related to the WordPress app, will be removed when we're ready to deploy all web-related changes as part of the Jetpack app migration.
* This PR also includes some light refactoring of the HTML and CSS for the `app-promo` component, to make sure of `display: flex` for positioning elements, rather than `position: absolute`. This was done to make it easier to match the designs for promoting the mobile app.
* As the `app-promo` component is used to promote both the mobile and the desktop app, these changes modifies the design for both promotions. The screenshots below show all before/after states.
</details>

#### Screenshots

<details>
 <summary>Design comparison ⤵️</summary>

| Design  | Actual |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/192336752-f8504e0e-41ef-450b-93e4-4773af5dabf9.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194160684-05a620f0-2abb-4825-94e3-5f8cf0b8020b.png" width="100%"> |
</details>

<details>
 <summary>Mobile app promo before/after ⤵️</summary>

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/192349308-aa14a9ab-1026-4df4-82c7-d8d67325b97d.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194160684-05a620f0-2abb-4825-94e3-5f8cf0b8020b.png" width="100%"> |
</details>

<details>
 <summary>Desktop app promos before/after ⤵️</summary>

As this change involved some changes to the promo, the design for the desktop app promo has been tweaked slightly:

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/192349153-58675fd1-c268-466a-8755-059f6f76b769.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194160009-822bf9e6-0f57-42b7-ae08-8ad5ed13157d.png" width="100%"> |

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/192348982-c2095b68-0602-4222-b465-a52ce6975c42.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194160153-86ec3754-e9e6-4e10-91f8-b834349bc855.png" width="100%"> |

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/192349896-39621899-4af9-4d41-a7e3-fee5acd0ea98.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194160494-d6eb96ad-afd5-48a1-b97e-f55329a7c7b9.png" width="100%"> |

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/192349592-7b4b0170-d854-413b-a51f-01461b635c37.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194160962-058f4897-8e6a-4209-8daa-62ec9080543f.png" width="100%"> |

</details>

<details>
 <summary>Different colour schemes ⤵️</summary>

**Ectoplasm**

| Desktop  | Mobile |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/194352818-b4390a15-4bb4-48e9-8f2b-30a3cb70de86.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194352676-c261cdfa-7513-4cc5-b559-669a2aa3c0df.png" width="100%"> |

**Nightfall**

| Desktop  | Mobile |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/194352461-48a429cc-811d-415e-94de-566743483ef5.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194352280-4935e4fc-9a25-451c-835a-c04d64dddc13.png" width="100%"> |

**Ocean**

| Desktop  | Mobile |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/194353171-f282eee6-baaf-464d-bbd7-93ebddb99d30.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194353356-663c1c56-bbed-442f-859b-10c897d1e92c.png" width="100%"> |

**Sunrise**

| Desktop  | Mobile |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/194353779-2ede12d8-0946-44d2-8fc3-8ab38d4e47fb.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194354300-6f8d2d09-e576-42f3-9133-75831cf3bbe1.png" width="100%"> |

**Sunset**

| Desktop  | Mobile |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/194352077-134f67af-fd4a-4bca-a91c-b40ded91527c.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194354443-97b4fec0-e538-409b-b045-d55a3153c98e.png" width="100%"> |

</details>


#### Testing Instructions

* Navigate to the Reader in an account where you haven't dismissed the sidebar app promo. (You can manually change `shouldRenderAppPromo` to `true` [here](https://github.com/Automattic/wp-calypso/blob/dded2cf2a4d3d3770818998875bd4c3616bd93f7/client/reader/sidebar/promo.jsx#L17) if testing locally.)
* Reload the page to cycle through the various app promos to confirm their appearance matches the above screenshots.

#### Pre-merge Checklist






Related to #
